src/modules/routes/Routes.ts

```diff
@@ -36,10 +42,21 @@ export default class Routes {
         return this.regionRoutes[normalizedRoute - 1].number;
     }
 
-    public static normalizedNumber(region: GameConstants.Region, route: number): number {
+    public static normalizedNumber(region: GameConstants.Region, route: number, skipIgnoredRoutes: boolean): number {
         if (region === GameConstants.Region.none) {
             return route;
         }
-        return this.regionRoutes.findIndex((routeData) => routeData.region === region && routeData.number === route) + 1;
+        const filteredRegionRoutes = this.regionRoutes.filter((r) => !skipIgnoredRoutes || !r.ignoreRouteInCalculations);
+        if (skipIgnoredRoutes && this.regionRoutes.find((routeData) => routeData.region === region && routeData.number === route)?.ignoreRouteInCalculations) {
+            for (let i = this.regionRoutes.findIndex((routeData) => routeData.region === region && routeData.number === route) - 1; i >= 0; i--) {
+                if (!this.regionRoutes[i].ignoreRouteInCalculations) {
+                    return i + 1;
+                }
+                if (i === 0) {
+                    throw new Error('Not implemented for ignoreRouteInCalculations = true on first region route');
+                }
+            }
+        }
+        return filteredRegionRoutes.findIndex((routeData) => routeData.region === region && routeData.number === route) + 1;
```

Comment:
Could we please get some code comments added here, as it's a bit complicated what's happening